qt/python/mantidqtinterfaces/mantidqtinterfaces/Engineering/gui/engineering_diffraction/tabs/fitting/plotting/plot_model.py

```diff
@@ -80,12 +85,38 @@ def update_fit(self, fit_props, loaded_ws_list, active_ws_list, log_workspace_na
                             self._fit_results[wsname]["results"][key_d].append([dcen, dcen_er])
                         except (ValueError, RuntimeError) as e:
                             logger.warning(f"Unable to output {key_d} parameters for TOF={params_dict['Value'][irow]}: " + str(e))
+                self._calculate_fwhm_values(
+                    ws_name=wsname, func_name=fname, numb_func_params=nparams[ifunc], params_dict=params_dict, params_start_index=istart
+                )
                 istart += nparams[ifunc]
             # append the cost function value (in this case always chisq/DOF) as don't let user change cost func
             # always last row in parameters table
             self._fit_results[wsname]["costFunction"] = params_dict["Value"][-1]
         self.create_fit_tables(loaded_ws_list, active_ws_list, log_workspace_name)
 
+    def _calculate_fwhm_values(self, ws_name, func_name, numb_func_params, params_dict, params_start_index):
```

Comment:
I wonder whether it would be easier to pass the function object here rather than the name and the `params_dict` (having created it from the whole function string - and if composite function split into sub-functions). You can't cast the function object to a peak function so you'll still have to use `peak_func = FunctionFactory.Instance().createPeakFunction(func_name)` and then something like `[peak_func.setParameter(iparam, func.getParameterValue(iparam)) for iparam in range(func.nParams())]` and probably the same for attributes using `peak_func.setAttributeValue`.